druid-shell/src/backend/windows/window.rs

```diff
@@ -1882,9 +1883,17 @@ impl WindowHandle {
         self.defer(DeferredOp::ShowTitlebar(show_titlebar));
     }
 
-    // Sets the position of the window in virtual screen coordinates
     pub fn set_position(&self, position: Point) {
         self.defer(DeferredOp::SetWindowState(window::WindowState::Restored));
+        let position = unsafe {
+            if !GetWindow(self.get_hwnd().unwrap(), GW_OWNER).is_null() {
```

Comment:
@jneem 
I think it would be better to store `WindowLevel` in `WindowHandle`. 
We would then be consistent with what is done in `build()`. ie convert point from window coords to screen coords only when this subwindow is `WindowLevel::Tooltip` or `WindowLevel::Dropdown`. 

What do you think?

Reply:
There's a `set_level` for `WindowHandle` which is unimpl today. Maybe above suggestion will fix that too.

Reply:
Hm, I can see the argument for including `WindowLevel` in `WindowHandle`, but it will require cross-platform changes and I don't think there's a *big* advantage, is there? I mean, you can query the parent easily enough from the windows API (and I think on other platforms too).

Also, I thought we had agreed in #1824 to remove `set_level` from `WindowHandle`.

Reply:
Few points:
* My point is consistency between `build()` and `set_position()`. Since `build()` calculates positions based on `WindowLevel` so should `set_position`. And not make a platform call which many might not know straight away. (it also negates the need for an extra `unsafe` call if that matters)
* Also there's a bug in `set_position`. I cant use `self.get_position()`. I need parent's position not its own. We can get that if `WindowLevel` (which internally gives us `WindowHandle`) is stored in `WindowHandle`
* To further my previous point, in gtk, parent handle is stored in `WindowHandle` for exactly that purpose. Only difference is _I think_ it's better to store `WindowLevel` (again we get window handle internally) for the sake of first point above. 

_I keep saying store `WindowLevel` in `WindowHandle` but what I actually mean is store it within one of its fields. Like in Windows can store level in `WindowState` which is field of `WindowHandle`_

> Also, I thought we had agreed in #1824 to remove `set_level` from `WindowHandle`.

Ohh. You are right. I can remove it as part of this?

Reply:
> I keep saying store `WindowLevel` in `WindowHandle` but what I actually mean is store it within one of its fields.

Ah, got it. I thought you were proposing to put it in the actual cross-platform `WindowHandle` struct, which would then require more invasive changes...

Can you explain the benefit of storing the `WindowLevel` as opposed to just the parent? My understanding of the position calculation was that it should only depend on whether there's a parent or not.

Reply:
> Ohh. You are right. I can remove it as part of this?

Please!

Reply:
> Can you explain the benefit of storing the WindowLevel as opposed to just the parent?

So instead of doing this:
```
if let Some(p) = self.parent {
    // use p to calculate position
}
```

We can do this:
```
if WindowLevel::Tooltip(p) || WindowLevel::Dropdown(p) {
    // use p to calculate position
}
```

`WindowLevel` check is consistent with `build()`

Reply:
...but I think you also want to include `WindowLevel::Modal(p)` there, because the only difference between those 3 levels is their z-order. They should all treat positioning in the same way.

Reply:
True. But at least in Windows it is unimplemented/unfinished. I didnt include it because I that change should be included when modal feature is implemented.

Reply:
Not sure if you agree to my previous point but Im going to resolve this conversation. It's been a while. If you disagree, please let me know and I'll add `WindowLevel::Modal(p)`.

Reply:
Oh, sorry, I had forgotten about this, but I don't understand the part about modal not being implemented in windows. The only difference between (druid terminology) tooltip, modal, and dropdown is that they have different z-order. They should have exactly the same behavior in all other respects, and so there's no reason to support, say, dropdown without also supporting modal.

Reply:
Where is the best place in code to see different z-orders in action? In Windows at least, z-order is same for tooltips and dropdown (and maybe for modal) ie 1 + max of existing z-orders.

Reply:
It seems like the mac backend is the only one that handles `WindowLevel` according to the documentation (see `as_raw_window_level`), but I don't have a mac so I can't test it. The gtk backend does something different. I guess what happened was that `WindowLevel` was created according to the mac behavior, and then because the variants have suggestive-sounding names it was implemented differently on gtk (and partially on windows).

But I think the right way forward is the following: let's first make window parent/child positioning work. After that, we can worry about z-order and any other behavioral changes that different subwindows can have. We'll need to figure out what we can reliably support cross-platform.

Reply:
Sounds good!!
I have made changes as per this conversation.